### ISSUE TITLE
Softmax over neighbors

### ISSUE THREAD

**Issue by Mctigger** [rusty1s/pytorch_geometric#532]
Hey,
probably a simple question :)

Given `def message(self, x_i, edge_index)` how can softmax along all neighbors of a node?

**Comment by rusty1s** [comment]
You can see it in action [here](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/conv/gat_conv.py#L103).
In the `message` function you need access to the neighboring node features `x_j`, and the target nodes to which those features get passed to (`edge_index_i`). The `size_i` argument denotes the number of nodes in your graph (and more concretely the number of receiving nodes in a bipartite graph):
```python
from torch_geometric.utils import softmax
alpha = softmax(alpha, edge_index_i, num_nodes)
```

**Comment by Mctigger** [comment]
Makes sense. Thanks a lot!
